tt_metal/impl/device/device_pool.cpp

```diff
@@ -314,13 +315,12 @@ bool DevicePool::is_device_active(chip_id_t id) const {
 }
 
 void DevicePool::add_devices_to_pool(const std::vector<chip_id_t>& device_ids) {
+    std::set<chip_id_t> devices_to_activate;
     if (this->skip_remote_devices) {
         for (const auto& device_id : device_ids) {
             const auto& mmio_device_id = tt::Cluster::instance().get_associated_mmio_device(device_id);
             TT_ASSERT(device_id == mmio_device_id, "Skipping remote devices is only available for mmio devices");
-            if (not this->is_device_active(device_id)) {
-                this->activate_device(device_id);
-            }
+            devices_to_activate.insert(device_id);
```

Comment:
This changes behaviour, are we always re-initializing devices now?

Reply:
I did want to open a conversation about the expected usage of this actually. If we want to pull fd init into the DevicePool going forward, we'll likely need to set some expectations on how this function is called.

As things stand right now, we need to have all devices added together here in order to generate the correct dispatch architecture, hence the always re-initialization. Do we think that users will want to initialize the fd architecture in pieces, or will a one-shot setup be the expected flow?

I also spoke to @cfjchu about allowing individual devices to be opened/closed for Galaxy, which would require the next step of pulling all of the FD init into DevicePool - we'll need to decide how we want the flow to look with respect to that as well